roles/openshift_health_checker/openshift_checks/memory_availability.py

```diff
@@ -0,0 +1,31 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException, get_var
+
+
+class MemoryAvailability(OpenShiftCheck):
+    """Check that recommended memory is available."""
+
+    name = "memory_availability"
+    tags = ["preflight"]
+
+    recommended_memory_mb = {
+        "nodes": 8 * 1024,
+        "masters": 16 * 1024,
+        "etcd": 10 * 1024,
+    }
+
+    def run(self, tmp, task_vars):
+        group_names = get_var(task_vars, "group_names", default=[])
+        total_memory = get_var(task_vars, "ansible_memtotal_mb")
+
+        recommended_memory_mb = max(self.recommended_memory_mb.get(group, 0) for group in group_names)
+        if not recommended_memory_mb:
+            msg = "Unable to determine recommended memory size for group_name {group_name}"
+            raise OpenShiftCheckException(msg.format(group_name=group_names))
+
+        if total_memory < recommended_memory_mb:
+            msg = ("Available memory ({available} GB) below recommended storage. "
+                   "Minimum required memory: {recommended} GB")
+            return {"failed": True, "msg": msg.format(available=total_memory, recommended=recommended_memory_mb)}
```

Comment:
The units are mismatched... we say "GB" in `msg`, but the variables are in MB...